### PR TITLE
Remove Airlift library usage from clients

### PR DESCRIPTION
**Pull request by electrum**
This allows Airlift libraries to be updated to Java 11.

### CODE REVIEW

File: client/trino-client/src/main/java/io/trino/client/JsonCodec.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+
+import java.lang.reflect.Type;
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+public class JsonCodec<T>
```
Comment (dain) [inline client/trino-client/src/main/java/io/trino/client/JsonCodec.java:28]: add the copy of note

Comment (dain) [reply client/trino-client/src/main/java/io/trino/client/JsonCodec.java:28]: Can we make this package protected?

**Review by dain — APPROVED** [review]
Minor comment but otherwise looks goo

File: client/trino-client/src/main/java/io/trino/client/JsonCodec.java (hunk 1 above)
Comment (electrum) [reply client/trino-client/src/main/java/io/trino/client/JsonCodec.java:28]: It's part of the public API of `JsonResponse` which is used by the `io.trino.client.auth.external` package.

**Pull request merged**